File: RoundManager.java
import java.awt.Color;

public abstract class RoundManager {
   private RoundManager prevRound;
   static final Color[] colors;
   protected int nextRoundScore;
   protected Color skyColor;
   protected Color groundColor;
   protected int gameTime;

   protected final Obstacle createObstacle(GameRecorder var1, double var2, double var4) {
      Obstacle var6 = Obstacle.newObstacle();
      DPoint3[] var7 = var6.points;
      var7[0].setXYZ(var2 - var4, 2.0D, 25.5D);
      var7[1].setXYZ(var2, -1.4D, 25.0D);
      var7[2].setXYZ(var2 + var4, 2.0D, 25.5D);
      var7[3].setXYZ(var2, 2.0D, 24.5D);
      var6.color = colors[var1.getRandom() % 4];
      var6.prepareNewObstacle();
      return var6;
   }

   protected final Obstacle createObstacle(GameRecorder var1, double var2) {
      double var4 = (double)(var1.getRandom() % 256) / 8.0D - 16.0D;
      return this.createObstacle(var1, var4, 0.6D);
   }

   public abstract void generateObstacle(ObstacleCollection var1, GameRecorder var2);

   public boolean isNextRound(int var1) {
      return var1 >= this.nextRoundScore;
   }

   public void setPrevRound(RoundManager var1) {
      this.prevRound = var1;
   }

   static {
      colors = new Color[]{Color.lightGray, new Color(96, 160, 240), new Color(200, 128, 0), new Color(240, 210, 100)};
   }

   public Color getGroundColor() {
      return this.groundColor;
   }

   public int getNextRoundScore() {
      return this.nextRoundScore;
   }

   public void init() {
   }

   public Color getSkyColor() {
      if (this.prevRound != null && this.gameTime <= 32) {
         int var1 = this.gameTime;
         int var2 = 32 - var1;
         Color var3 = this.prevRound.skyColor;
         int var4 = var3.getRed() * var2 + this.skyColor.getRed() * var1;
         int var5 = var3.getGreen() * var2 + this.skyColor.getGreen() * var1;
         int var6 = var3.getBlue() * var2 + this.skyColor.getBlue() * var1;
         return new Color(var4 / 32, var5 / 32, var6 / 32);
      } else {
         return this.skyColor;
      }
   }

   public void move(double var1) {
   }
}

File: js/roundManager.js
import { Obstacle } from './obstacle.js';

// Obstacle colors matching the original
export const OBSTACLE_COLORS = [
  { r: 192, g: 192, b: 192 }, // lightGray
  { r: 96, g: 160, b: 240 },
  { r: 200, g: 128, b: 0 },
  { r: 240, g: 210, b: 100 }
];

export class RoundManager {
  constructor(nextRoundScore, skyColor, groundColor) {
    this.nextRoundScore = nextRoundScore;
    this.skyColor = skyColor;
    this.groundColor = groundColor;
    this.gameTime = 0;
    this.prevRound = null;
  }

  createObstacle(recorder, x, width) {
    const ob = Obstacle.newObstacle();
    const p = ob.points;
    p[0].setXYZ(x - width, 2.0, 25.5);
    p[1].setXYZ(x, -1.4, 25.0);
    p[2].setXYZ(x + width, 2.0, 25.5);
    p[3].setXYZ(x, 2.0, 24.5);
    ob.color = OBSTACLE_COLORS[recorder.getRandom() % 4];
    ob.prepareNewObstacle();
    return ob;
  }

  createObstacleRandom(recorder) {
    const x = (recorder.getRandom() % 256) / 8.0 - 16.0;
    return this.createObstacle(recorder, x, 0.6);
  }

  generateObstacle(obstacles, recorder) {
    // Abstract - override in subclasses
  }

  isNextRound(score) {
    return score >= this.nextRoundScore;
  }

  getNextRoundScore() {
    return this.nextRoundScore;
  }

  setPrevRound(prev) {
    this.prevRound = prev;
  }

  init() {
    // Override in subclasses
  }

  getGroundColor() {
    return this.groundColor;
  }

  getSkyColor() {
    if (this.prevRound !== null && this.gameTime <= 32) {
      const t = this.gameTime;
      const t2 = 32 - t;
      const prev = this.prevRound.skyColor;
      const r = ((prev.r * t2 + this.skyColor.r * t) / 32) | 0;
      const g = ((prev.g * t2 + this.skyColor.g * t) / 32) | 0;
      const b = ((prev.b * t2 + this.skyColor.b * t) / 32) | 0;
      return { r, g, b };
    }
    return this.skyColor;
  }

  move(vx) {
    // Override in subclasses if needed
  }
}
